### PR TITLE
Create ht4.go

### PR DESCRIPTION
**Pull request by slav4ik-py**

### CODE REVIEW

File: ht4/ht4.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		if len(i) > maxLength {
+			maxLength = len(i)
+			stringWithMaxLength = i
+		}
+	}
+	return stringWithMaxLength
+}
+
+func reverse(array []int64) []int64 {
+	var resultSlice []int64
```
Comment (wshaman) [inline ht4/ht4.go]:
```suggestion
	resultSlice := make([]int64, 0, len(array))
```

Comment (wshaman) [reply ht4/ht4.go]: This change allows to keep slice capacity equal to it's len

File: ht4/ht4.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package ht4
+
+func max(array []string) string {
```
Comment (wshaman) [inline ht4/ht4.go:3]: This is not the best name for func. eg `longestWord` would be more self explanatory

File: ht4/ht4.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+func reverse(array []int64) []int64 {
+	var resultSlice []int64
+	for i := len(array) - 1; i >= 0; i-- {
+		resultSlice = append(resultSlice, array[i])
+	}
+	return resultSlice
+}
+
+// CallMax : This method is used for check 'max' of ht3 in 'main.go'
+func CallMax(array []string) string {
```
Comment (wshaman) [inline ht4/ht4.go:24]: Do we really need this separated func?

**Review by wshaman — APPROVED** [review]

File: ht4/ht4.go (hunk 2 above)
Comment (slav4ik-py) [reply ht4/ht4.go:3]: This function has name "max" because this name was marked in description of the task

File: ht4/ht4.go (hunk 3 above)
Comment (slav4ik-py) [reply ht4/ht4.go:24]: I use this function in main.go

**Review by ikamornikau — APPROVED** [review]
LGTM

**Pull request merged**